### ISSUE TITLE
 Clean up lombok annotations in API response models

### ISSUE THREAD

**Issue by wslulciuc** [MarquezProject/marquez#355]
We can clean up and only annotate our API **response** models with the following lombok annotations: [`@EqualsAndHashCode`](https://projectlombok.org/features/EqualsAndHashCode) and [`@ToString`](https://projectlombok.org/features/ToString)

**For example:**

```java
@EqualsAndHashCode
@ToString
public final class FooResponse {
  @Getter private final String value;
  private final String description;

  public FooResponse(
      @NonNull final String value,
      @Nullable final String description) {
    this.name = checkNotBlank(value);
    this.description = description;
  }

  public Optional<String> getDescription() {
    return Optional.ofNullable(description);
  }
}
```

**NOTE:** For a great API response model example, see [`DatasetResponse`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/api/models/DatasetResponse.java)

**Comment by hjpatel16** [comment]
Hey, can i work on it?

**Comment by wslulciuc** [comment]
Sure! Thanks, @hjpatel16

**Comment by wslulciuc** [comment]
@hjpatel16: Did you still want to work on this issue? Let us know :)

**Comment by ravikamaraj** [comment]
Hi @wslulciuc, Shall I pick this issue

**Comment by wslulciuc** [comment]
@ravikamaraj, sorry for the delayed response. Sure, thanks for picking up this issue!

**Comment by ravikamaraj** [comment]
@wslulciuc , np. thanks

**Comment by ravikamaraj** [comment]
@wslulciuc , can you close this issue. thanks. 

PS: _with recent PR's, I am following  CONTRIBUTING.md  to close the issue automatically upon PR approval_

**Comment by wslulciuc** [comment]
Fixed #573
